Issue #1770

dnahodil opened the pull request:
Fixes #1770

jkburges commented on src/test/javascript/portal/ui/openlayers/control/SpatialConstraintSpec.js:
```diff
@@ -309,22 +308,18 @@ describe('Portal.ui.openlayers.control.SpatialConstraint', function() {
             expect(spatialConstraint._checkSketch(feature)).toEqual(true);
         });
 
-        it('returns false if geometry is too small, shows error', function() {
```
![image](https://cloud.githubusercontent.com/assets/1849176/8223312/93bef142-15b9-11e5-9617-9d0f1f4d5bcb.png)

dnahodil replied:
The _checkSketch() method previously checked the validity of the geometry *and* called the _showAnteMeridianError(). Now it's only job is to return the validity.

But I now realise I didn't write tests to check my new behaviours so I will do that.

jkburges commented on web-app/js/portal/ui/openlayers/control/SpatialConstraint.js:
```diff
@@ -13,7 +13,7 @@ Portal.ui.openlayers.SpatialConstraintType = {
 
```
This class is quite big, and complex, IMO.  What are its main responsibilities currently (I could answer that but since you've just looked at the code, perhaps you're in a better position to do so)?

dnahodil replied:
Good question.

It represents the control (a non-visible object) for the geometry you can draw on the map.
There is code to:
* Attach / remove it from the map
* Initialise its events
* Get the constraint / geometry that has been drawn
* Arrange the layer showing the geometry on the map
* Validating the geometry that is drawn
* Visualising an error if it occurs

So it does do a lot of things. I will have a think which of those responsibilities could be nicely extracted.

dnahodil commented:
@jkburges review changes have been made. I've added some more tests for the functionality I added and I extracted the logic for validating the drawn geometry. There are probably other candidates for new classes to be extracted from the SpatialConstraint but I think we should leave them for another day.

jkburges commented:
:+1: it's a step in the right direction.
